validator/sawtooth_validator/execution/scheduler_serial.py

```diff
@@ -186,11 +186,14 @@ def _compute_merkle_root(self, required_state_root):
         return state_hash
 
     def _calculate_state_root_if_not_already_done(self):
-        last_txn_signature = self._last_in_batch[-1]
-        batch_id = self._txn_to_batch[last_txn_signature]
-        required_state_hash = self._required_state_hashes.get(
-            batch_id)
         if not self._already_calculated:
+            if not len(self._last_in_batch) > 0:
```

Comment:
To follow along with the new lint PR, this should probably be "if not self._last_in_batch:"